src/VirtualClient/VirtualClient.Actions/DCGMI/DCGMIProftester/DCGMIProftesterCommandParser.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Text;
+    using System.Text.Json;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.Identity.Client;
+    using VirtualClient.Contracts;
+    using YamlDotNet.Core.Tokens;
+    using DataTableExtensions = VirtualClient.Contracts.DataTableExtensions;
+
+    /// <summary>
+    /// Parser for DCGMI Proftester output document.
+    /// </summary>
+    public class DCGMIProftesterCommandParser : MetricsParser
```

Comment:
Referencing previous comments on consolidating the logic of the parsers. The parsers for Proftester are an example of where we can probably just put the logic in methods within a single parser. There is a cost to having too many abstractions in code. When the code is distinctively different, then it warrants placing the logic in separate classes and incurring the cost of additional abstractions. When the logic is doing very similar things (e.g. parsing DCGMI output), the cost of the abstractions should be avoided.